truffle/com.oracle.truffle.api.test/src/com/oracle/truffle/api/vm/EngineTest.java

```diff
@@ -117,4 +119,19 @@ public void wrappedAsArray() throws Exception {
         assertEquals(2, arr[1].intValue());
         assertEquals(3, arr[2].intValue());
     }
+
+    @Test
+    public void initializePolyglotEngineWithArguments() throws IOException {
+        PolyglotEngine vm = createBuilder().setArguments(ExportImportLanguage1.class, new String[]{"1", "2"}).build();
```

Comment:
Alas, I don't think people should reference the implementation class of a language. They certainly don't have to do so now. Probably the language should be identified by a MIME type.